paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -117,6 +137,120 @@ internal class PaymentSheetPlaygroundViewModel(
         }
     }
 
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private fun prepareCustomer(
+        playgroundSettings: PlaygroundSettings
+    ) {
+        viewModelScope.launch {
+            val snapshot = playgroundSettings.snapshot()
+
+            state.value = PlaygroundState.Customer(
+                snapshot = snapshot,
+                adapter = CustomerAdapter.create(
+                    context = getApplication(),
+                    customerEphemeralKeyProvider = ::fetchEphemeralKey,
+                    setupIntentClientSecretProvider = if (
+                        snapshot[CustomerSheetPaymentMethodModeDefinition] == PaymentMethodMode.SetupIntent
+                    ) {
+                        { createSetupIntentClientSecret(it) }
+                    } else {
+                        null
+                    }
+                )
+            )
+        }
+    }
+
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private suspend fun fetchEphemeralKey(): CustomerAdapter.Result<CustomerEphemeralKey> {
+        val playgroundSettings = playgroundSettingsFlow.value
+            ?: return CustomerAdapter.Result.failure(
+                cause = Exception("Unexpected error: Playground settings were not found!"),
+                displayMessage = null,
+            )
+
+        // Snapshot before making the network request to not rely on UI staying in sync.
+        val playgroundSettingsSnapshot = playgroundSettings.snapshot()
```

Comment:
I think we need to pass in the already snapshotted settings, rather than create it again.